src/Word/CamelCaseToSeparator.php

```diff
@@ -15,23 +16,24 @@
  *     ...
  * }
  * @template TOptions of Options
- * @extends AbstractSeparator<TOptions>
+ * @implements FilterInterface<string|array<array-key, string|mixed>>
  */
-class CamelCaseToSeparator extends AbstractSeparator
+class CamelCaseToSeparator implements FilterInterface
 {
-    public function filter(mixed $value): mixed
+    private readonly string $separator;
+
+    /** @param Options $options */
+    public function __construct(array $options = [])
     {
-        return self::applyFilterOnlyToStringableValuesAndStringableArrayValues(
-            $value,
-            Closure::fromCallable([$this, 'filterNormalizedValue'])
-        );
+        $this->separator = $options['separator'] ?? ' ';
+    }
+
+    public function __invoke(mixed $value): mixed
+    {
+        return $this->filter($value);
     }
 
-    /**
-     * @param  string|string[] $value
-     * @return string|string[]
-     */
-    private function filterNormalizedValue(string|array $value): string|array
+    public function filter(mixed $value): mixed
     {
         if (StringUtils::hasPcreUnicodeSupport()) {
```

Comment:
All of this can be dropped - PHP has had PCRE unicode support since 5.1 IIRC

If you look at [issue 10](https://github.com/laminas/laminas-filter/issues/10) - there are some problems with the pattern too which I think should be fixed

```php
$pattern = '#([\p{Lu}][\p{Ll}]+)#';
$result = implode($separator, preg_split($pattern, $subject, -1, PREG_SPLIT_DELIM_CAPTURE|PREG_SPLIT_NO_EMPTY));
```

👆 That might do the trick, but it doesn't account for existing non-word chars.